app/helpers/submissions_helper.rb

```diff
@@ -38,6 +38,18 @@ def safe_submission_avatar(submission)
     submission.drawing.avatar.url
   end
 
+  def safe_submission_drawing(submission)
+    unless logged_in?
+      return NSFW_THUMB if submission.nsfw_level > 1
+
+      return submission.drawing.avatar.url
+    end
+
+    return NSFW_THUMB if current_user.nsfw_level < submission.nsfw_level
+
+    submission.drawing.url
+  end
```

Comment:
I would've said to make this a method based on the Submission class, but there's some weird stuff with regards to session helpers (so the models don't have access to logged_in? nor current_user).
I'll see if there's a good way to refactor this in the future, but for now this is fine.

Reply:
also I didn't see any hi res NSFW_DRAWING so when we click on a NSFW drawing it will appear as a pixelated image inside the modal